restate-test-utils/src/test_utils.rs

```diff
@@ -0,0 +1,174 @@
+            .await?
+            .json::<VersionResponse>()
+            .await?;
+
+        println!("body:\n{:?}", body);
+
+        Ok(TestContainer {container})
+    }
+
+    pub async fn register(&self, server_port:u16) -> Result<(), HandlerError> {
+
+        let host = self.container.get_host().await?;
+        let ports = self.container.ports().await?;
+
+        let admin_port = ports.map_to_host_port_ipv4(9070.tcp()).unwrap();
+        let server_url = format!("http://localhost:{}", admin_port);
+
+        println!("health check:       {}/health", server_url);
+
+        let client = reqwest::Client::builder().http2_prior_knowledge().build().unwrap();
+
+        // wait for server to respond
+        while let Err(err) = client.get(format!("{}/health", server_url).to_string())
+                                .header("accept", "application/vnd.restate.endpointmanifest.v1+json")
+                                .send().await {
+            tokio::time::sleep(Duration::from_secs(1)).await;
+
+            println!("health check:       {:?}", err);
+        }
+
+        let ok = client.get(format!("{}/health", server_url).to_string())
+                                .header("accept", "application/vnd.restate.endpointmanifest.v1+json")
+                                .send().await;
+
+        println!("health check:       {:?}", ok);
+
+        println!("host started");
+
+        let uri = http::Uri::builder()
+            .scheme("http")
+            .authority(format!("host.docker.internal:{}", server_port))
+            .path_and_query("/")
+            .build().unwrap();
+
+        let register_url = format!("http://{}:{}/deployments", host, admin_port);
+        println!("uri_str:            {:?}", uri);
+
+        let register_payload = RegisterDeploymentRequest::Http { uri, additional_headers: None, use_http_11: false, force: false, dry_run: false }; //, additional_headers: (), use_http_11: (), force: (), dry_run: () }
+        //let register_payload = format!("{{\"uri\": \"host.docker.internal:{}/\", \"force\": true}}", server_port);
+
+        println!("register payload:   \n{:?}", register_payload);
+
+        // wait for registration
+        let registed = client.post(register_url.to_string())
+                            .json(&register_payload)
+                            .send().await;
+
+        assert!(registed.is_ok());
+
+        let response =registed.unwrap();
+        assert!(response.status().is_success());
+
+        println!("register response:  {}\n{}", nu_ansi_term::Style::new().bold().paint(response.status().as_u16().to_string()), response.text().await.unwrap());
+        println!("{}", nu_ansi_term::Style::new().bold().paint("host registered"));
+
+        return Ok(());
+    }
+
+    pub async fn delay(milliseconds:u64) {
+        tokio::time::sleep(Duration::from_millis(milliseconds)).await;
+    }
+
+    pub async fn invoke(&self, service:Service, handler:String) -> Result<(), HandlerError> {
```

Comment:
Rather than this, i would start the user services http server and register them when bootstrapping the test infra, then just provide a `url` here, and let people use the client they want to send test requests